services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/ScanOperationTest.java

```diff
@@ -188,6 +189,25 @@ public void generateRequest_consistentRead() {
         assertThat(request, is(expectedRequest));
     }
 
+    @Test
```

Comment:
Could we test a filter expression together with a projection expression and verify the merging logic is working correctly?